Add protocol details (hybrid topology, etc) to docs (#776)
OpenFreeEnergy/openfe: docs/guide/protocols/plainmd.rst

Plain MD Protocol
=================

Overview
--------

> The :class:`.PlainMDProtocol` enables the user to run an MD simulation of a ``ChemicalSystem``, which can contain e.g. a solvated protein-ligand complex, or a ligand and water.

Reviewer: or a ligand in vacuum? - ah I see it's mentioned below, might not be necessary!

Reviewer: ```suggestion
The :class:`.PlainMDProtocol` enables the user to run a Molecular Dynamics (MD) simulation of a ``ChemicalSystem``, which can contain e.g. a solvated protein-ligand complex, or a ligand and water.
```

Sorry this is probably pedantic, but it's the first time you're defining it here I think?

TODO: Later add ref to ChemicalSystem section

Scientific Details
------------------

The :class:`.PlainMDProtocol` runs MD simulations of a system either in solvent or vacuum, depending on the input provided by the user in the `ChemicalSystem`.
The protocol applies a 
`LangevinMiddleIntegrator <http://docs.openmm.org/development/api-python/generated/openmm.openmm.LangevinMiddleIntegrator.html>`_ 
which uses Langevin dynamics, with the LFMiddle discretization [1]_.  

Simulation Steps
~~~~~~~~~~~~~~~~

If there is a ``SolventComponent`` in the ``ChemicalSystem``, the each Protocol simulation Unit carries out the following steps:

1. Parameterize the system using `OpenMMForceFields <https://github.com/openmm/openmmforcefields>`_ and `Open Force Field <https://github.com/openforcefield/openff-forcefields>`_.
2. Minimize the system

> 3. Equilibrate in the canonical ensemble
> 4. Equilibrate and production simulate the system (under NPT conditions using a MonteCarloBarostat to maintain constant pressure)

Reviewer: ```suggestion
3. Equilibrate in the canonical (NVT) ensemble
4. Equilibrate and production simulate the system under isobaric-isothermal (NPT) conditions using a MonteCarloBarostat to maintain constant pressure
```
😅 we don't necessary need to go this far, just if we wanted to spell out the ensembles we can do this

Relevant settings under solvent conditions include the solvation settings that control the ``solvent_model`` and ``solvent_padding``.

> If the ``ChemicalSystem`` does not contain a ``SolventComponent``, the protocol runs an MD simulation in vacuum. After a minimization, the protocol performs an NVT equilibration, followed by an NVT production run with no periodic boundary conditions and infinite cutoffs. Settings that control the barostat or the solvation are ignored for vaccum MD simulations.

Reviewer: ```suggestion
If the ``ChemicalSystem`` does not contain a ``SolventComponent``, the protocol runs an MD simulation in vacuum. After a minimization, the protocol performs an equilibration, followed by a production run with no periodic boundary conditions and infinite cutoffs. Settings that control the barostat or the solvation are ignored for vaccum MD simulations.
```
no pbc = no volume :)

Plain MD Protocol
=================

> The plain MD protocol enables the user to run an MD simulation of a `ChemicalSystem`, which can contain e.g. a solvated protein-ligand complex, ora ligand and water. 

Reviewer: Beyond my previous comment:

1. we should make sure that our first in-text reference to the Protocol points to the Protocol API (i.e. :class:`.PlainMDProtocol`.
2. Here we should talk about vacuum simulations too (since they are possible), this probably would fall under "simulation overview" or "scientific details" - something along the lines of "if you have a SolventComponent you will be running under these conditions and these settings are relevant. If you don't then you will be running in vacuum and you are expected to set this setting".

> The protocol applies a LangevinMiddleIntegrator which uses Langevin dynamics, with the LFMiddle discretization (J. Phys. Chem. A 2019, 123, 28, 6056-6079). 

Reviewer: I don't think we're using bibtext, so this would be a great one to throw in a reference section for now.
